XbSymbolDatabase.h

```diff
@@ -551,6 +951,73 @@ void XbSymbolContinuousSigScan(bool enable);
 /// <param name="enable">Input boolean to use first symbol address only or not.</param>
 void XbSymbolFirstDetectAddressOnly(bool enable);
 
+// ******************************************************************
+// * API v3 (context) functions to use with other projects.
+// ******************************************************************
+// NOTE: If need to make customize setup, please use the above functions which will be deprecate in the future.
+// before call XbSymbolContext_Create function.
+
+/// <summary>
+/// Step 1: Generate library array for LibraryHeader input.
+/// First call with <paramref name="library_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="library_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="library_out">Input null pointer will provide the count to allocate memory for library_out.filters field. Otherwise, information will be input to library_out.filters field.</param>
+unsigned int XbSymbolDatabase_GenerateLibraryFilter(const void* xb_header_addr, XbSDBLibraryHeader* library_out);
+
+/// <summary>
+/// Step 2: Generate section array for SectionHeader input.
+/// First call with <paramref name="section_out"/> as null pointer will return total count. Then second call will insert information to <paramref name="section_out"/>.filters field.
+/// </summary>
+/// <param name="xb_header_addr"></param>
+/// <param name="section_out">Input null pointer will provide the count to allocate memory, then allocate memory to section_out.filters field.</param>
+/// <param name="is_raw">True: Convert array to use raw xbe's relative address; False: Convert array to use xbox virtual memory relative address.</param>
+unsigned int XbSymbolDatabase_GenerateSectionFilter(const void* xb_header_addr, XbSDBSectionHeader* section_out, bool is_raw);
```

Comment:
This stuff (filters) could probably be treated similar to the function which creates a context (returning a handle and share the naming conventions: `Create` or something).

Basically this would imply that you have to create the filters first, in order to be able to create the context (as it depends on the filters). Then you can operate on the context.

So the API will eventually become self-documenting.

Reply:
I'm avoiding handle for both header's structures. Since I don't have a valid reason for one time usage which is only used in create process. And it's easier to reduce extra workload for simple allocation and set the new temporary buffer once. Once `XbSymbolContextCreate` have a copy of the filter's structure data. Then it is safe to free temporary buffer. Unless use the alternate function for multi-threading support, then you'll have to provide library info to `XbSymbolContext_ScanLibrary`.

Plus, I don't plan on implement a reference counter system.